src/test/chain_tests.cpp

```diff
@@ -41,6 +43,78 @@ const CBlockIndex* NaiveLastCommonAncestor(const CBlockIndex* a, const CBlockInd
 
 } // namespace
 
+BOOST_AUTO_TEST_CASE(get_skip_height_test)
+{
+    BOOST_CHECK(GetSkipHeight(0b0001011101101000) == 0b0001011101100000);
+    BOOST_CHECK(GetSkipHeight(0b0001011101101001) == 0b0001011101000001);
+    BOOST_CHECK(GetSkipHeight(0b0110101101011000) == 0b0110101101010000);
+    BOOST_CHECK(GetSkipHeight(0b0110101101011001) == 0b0110101101000001);
+    BOOST_CHECK(GetSkipHeight(0) == 0);
+    BOOST_CHECK(GetSkipHeight(1) == 0);
+    BOOST_CHECK(GetSkipHeight(2) == 0);
+    BOOST_CHECK(GetSkipHeight(3) == 1);
+    BOOST_CHECK(GetSkipHeight(4) == 0);
+    BOOST_CHECK(GetSkipHeight(5) == 1);
+    BOOST_CHECK(GetSkipHeight(7) == 1);
+    BOOST_CHECK(GetSkipHeight(8) == 0);
+    BOOST_CHECK(GetSkipHeight(15) == 9);
+    BOOST_CHECK(GetSkipHeight(16) == 0);
+    BOOST_CHECK(GetSkipHeight(17) == 1);
+    BOOST_CHECK(GetSkipHeight(256) == 0);
+    BOOST_CHECK(GetSkipHeight(65536) == 0);
+}
+
+BOOST_AUTO_TEST_CASE(skip_height_properties_test)
+{
+    // The test collects and analyses the distribution of the
+    // skip difference numbers, and checks some properties:
+    // non-uniform distribution with most values small but
```

Comment:
it seems to test more than non-uniform.

Reply:
I check the avg, which is computationally easier to obtain. It would make sense to check the median as well, and maybe compare the median with the average. However, I prefer to avoid the O(N log N) complexity of the sorting.

Reply:
Added median check as well. Much lower than the avg!